testing/unit/test_faucetizer.py

```diff
@@ -89,16 +169,137 @@ def tearDown(self):
+            ('02:00:00:00:00:01', {'switch': 't2sw2', 'port': 2, 'connected': True}, False),
+            ('02:00:00:00:00:03', {'switch': 't2sw2', 'port': 1, 'connected': True}, False),
+            # devices expired
+            ('02:00:00:00:00:01', {'switch': 't2sw2', 'port': 2, 'connected': False}, False),
+            ('02:00:00:00:00:03', {'switch': 't2sw2', 'port': 1, 'connected': False}, False)
+        ]
 
-        self.assertEqual(faucetizer_behavioral_config, expected_behavioral_config)
+        behaviors = [
+            # mocking static behaviors
+            ('02:00:00:00:00:01', {'segment': 'SEG_A', 'role': 'red'}, True),
+            ('02:00:00:00:00:03', {'segment': 'SEG_C', 'role': 'blue'}, True),
+            # devices authenticated
+            ('02:00:00:00:00:02', {'segment': 'SEG_B', 'role': 'green'}, False),
+            ('02:00:00:00:00:03', {'segment': 'SEG_A', 'role': 'yellow'}, False)
+        ]
+
+        # process static device info
+        self._process_device_placement(placements[0])
+        self._process_device_placement(placements[1])
+        self._process_device_behavior(behaviors[0])
+        self._process_device_behavior(behaviors[1])
+
+        # process dynamic device info
+        self._process_device_behavior(behaviors[2])
+        self._process_device_placement(placements[2])
+        self._process_device_placement(placements[3])
+        self._process_device_behavior(behaviors[3])
+
+        expected_behavioral_config_str = """
+        dps:
+          t1sw1:
+            dp_id: 111
+            interfaces:
+              1:
+                output_only: true
+              6:
+                stack: {dp: t2sw1, port: 6}
+              7:
+                stack: {dp: t2sw2, port: 7}
+              23:
+                lacp: 3
+          t2sw1:
+            dp_id: 121
+            interfaces:
+              1:
+                description: HOST
+                max_hosts: 1
+                native_vlan: 200
+                acls_in: [role_red, tail_acl]
+              2:
+                description: HOST
+                max_hosts: 1
+                native_vlan: 300
+                acls_in: [role_green, tail_acl]
+              6:
+                stack: {dp: t1sw1, port: 6}
+          t2sw2:
+            dp_id: 122
+            interfaces:
+              1:
+                description: HOST
+                max_hosts: 1
+                native_vlan: 400
+                acls_in: [role_blue, tail_acl]
+              2:
+                description: HOST
+                max_hosts: 1
+                native_vlan: 100
+              7:
+                stack: {dp: t1sw1, port: 7}
+        include: []
+        """
+        self._verify_behavioral_config(yaml.safe_load(expected_behavioral_config_str))
+
+        # device expired
+        self._process_device_placement(placements[4])
+        self._process_device_placement(placements[5])
+
+        expected_behavioral_config_str = """
```

Comment:
can you maybe just modify the expected_behavior_config dict (result from yaml.safe_load) from the previous bit with the explicit expected changes?  Having the complete representation of the structure in both places is tricky to read because it's hard to tell what's different!

Reply:
Got it. Added a class variable with basic behavioral config, and updating from there.